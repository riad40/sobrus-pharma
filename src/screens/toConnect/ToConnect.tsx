import React, { useState, useEffect } from 'react'
import { View, Text, SafeAreaView, ScrollView, Image } from 'react-native'
import { NavBar, Button } from '../../components'
import toConnectStyles from './ToConnect.styles'
import { useNavigation, NavigationProp } from '@react-navigation/native'
import RootStackParamsList from '../../navigations/RootStackParamsList'

const ToConnect = (): JSX.Element => {
    const [disabled, setDisabled] = useState<boolean>(false)
    const [loading, setLoading] = useState<boolean>(false)

    const navigation = useNavigation<NavigationProp<RootStackParamsList>>()

    const onPress = () => {
        setDisabled(true)
        setLoading(true)
        setTimeout(() => {
            navigation.navigate('Syncing')
        }, 2000)
    }

    useEffect(() => {
        const unscribe = navigation.addListener('focus', () => {
            setDisabled(false)
            setLoading(false)
        })
        return unscribe
    }, [])

    return (
        <SafeAreaView>
            <ScrollView style={toConnectStyles.screenContainer}>
                <NavBar />
                <View style={toConnectStyles.centeredImageContainer}>
                    <Image
                        testID="toConnectImage"
                        source={require('../../assets/images/toConnectImg.png')}
                        style={toConnectStyles.centeredImage}
                    />
                </View>
                <View style={toConnectStyles.bottomContent}>
                    <Text style={toConnectStyles.boldText}>Le logiciel qu’il vous faut</Text>
                    <Text style={toConnectStyles.lightText}>
                        Sobrus Pharma est un logiciel développé pour vous apporter une meilleure gestion de votre
                        officine.
                    </Text>
                </View>
                <Button text="Se Connecter" loading={loading} disabled={disabled} onPress={onPress} testID="Button" />
            </ScrollView>
        </SafeAreaView>
    )
}

export default ToConnect
