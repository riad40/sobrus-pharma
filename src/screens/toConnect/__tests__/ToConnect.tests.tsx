import React from 'react'
import ToConnect from '../ToConnect'
import { render, fireEvent, waitFor } from '@testing-library/react-native'
import { NavigationContainer, useNavigation } from '@react-navigation/native'

jest.useFakeTimers()

jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native')
    return {
        ...actualNav,
        useNavigation: jest.fn()
    }
})

describe('ToConnect Screen', () => {
    const mockNavigate = jest.fn()

    const mockUseNavigation = () => {
        return {
            navigate: mockNavigate,
            addListener: jest.fn()
        }
    }

    beforeEach(() => {
        ;(useNavigation as jest.Mock).mockReturnValue(mockUseNavigation())
    })

    const ToConnectScreen = (): JSX.Element => {
        return (
            <NavigationContainer>
                <ToConnect />
            </NavigationContainer>
        )
    }

    // test all the screen renders
    it('should render correctly', () => {
        const screen = render(<ToConnectScreen />)
        expect(screen).toMatchSnapshot()
    })

    // test the image renders
    it('should render the image', () => {
        const screen = render(<ToConnectScreen />)
        const image = screen.getByTestId('toConnectImage')
        expect(image).toBeTruthy()
    })

    // test the button press and wait 2 seconds then navigate to the next screen
    it('should navigate to the next screen', async () => {
        const screen = render(<ToConnectScreen />)
        const button = screen.getByTestId('Button')
        fireEvent.press(button)
        jest.advanceTimersByTime(2000)
        await waitFor(() => expect(mockNavigate).toHaveBeenCalledWith('Syncing'))
    })
})
