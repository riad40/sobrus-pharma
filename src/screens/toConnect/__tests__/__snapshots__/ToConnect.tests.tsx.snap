// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToConnect Screen should render correctly 1`] = `
<RCTSafeAreaView>
  <RCTScrollView
    style={
      {
        "backgroundColor": "#fff",
        "height": "100%",
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 30,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/images/pharmalogo.png",
            }
          }
          style={{}}
        />
        <Image
          source={
            {
              "testUri": "../../../src/assets/images/doctorHeadset.png",
            }
          }
          style={{}}
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/images/toConnectImg.png",
            }
          }
          style={
            {
              "height": 667,
              "resizeMode": "contain",
              "width": "100%",
            }
          }
          testID="toConnectImage"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 16,
              "marginBottom": 10,
            }
          }
        >
          Le logiciel qu’il vous faut
        </Text>
        <Text
          style={
            {
              "color": "#000",
              "fontFamily": "Poppins-Regular",
              "fontSize": 13,
              "marginBottom": 20,
            }
          }
        >
          Sobrus Pharma est un logiciel développé pour vous apporter une meilleure gestion de votre officine.
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#00C1B3",
            "borderRadius": 10,
            "opacity": 1,
            "paddingVertical": 20,
            "width": "90%",
          }
        }
        testID="Button"
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontFamily": "Poppins-Meduim",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Se Connecter
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
